### PR TITLE
Dnd sort order

### LINKED ISSUE #733: Team project columns (filtered by team member) cache/redux error on DnD
To reproduce:

1. Go to team dash
2. Filter by team member
3. Drag a project from one column to another
4. Error:

```Uncaught (in promise) Error: Cache went stale & wasn't recreated. Did you forget to put a redux subscriber on teamColumnsContainer?```

- **Estimated effort**: 12 points ([see CONTRIBUTING.md](https://github.com/ParabolInc/action/blob/master/CONTRIBUTING.md#points-and-sizes))

### LINKED ISSUE #778: Server exception on meetingSummary [dnd-sortOrder]
## Issue - Bug

Repro steps:

1. Login as `jordan@prbl.co`
2. Start meeting on `Parabol` team
3. Proceed through meeting:
   1. Create one agenda item
   2. Create two new output projects (1 for terry, 1 for jordan)
4. End meeting

Observe exception:

```
DEBUG GraphQL Error: [ { Error: Expected a value of type "ProjectStatus" but received: active
      at completeLeafValue (/Users/jordanhusney/Source/Repositories/github.com/Parabol/action.git/node_modules/graphql/execution/execute.js:639:11)
      at completeValue (/Users/jordanhusney/Source/Repositories/github.com/Parabol/action.git/node_modules/graphql/execution/execute.js:586:12)
      at completeValueWithLocatedError (/Users/jordanhusney/Source/Repositories/github.com/Parabol/action.git/node_modules/graphql/execution/execute.js:517:21)
      at completeValueCatchingError (/Users/jordanhusney/Source/Repositories/github.com/Parabol/action.git/node_modules/graphql/execution/execute.js:493:21)
      at resolveField (/Users/jordanhusney/Source/Repositories/github.com/Parabol/action.git/node_modules/graphql/execution/execute.js:461:10)
      at /Users/jordanhusney/Source/Repositories/github.com/Parabol/action.git/node_modules/graphql/execution/execute.js:273:18
      at Array.reduce (native)
      at executeFields (/Users/jordanhusney/Source/Repositories/github.com/Parabol/action.git/node_modules/graphql/execution/execute.js:270:42)
      at completeObjectValue (/Users/jordanhusney/Source/Repositories/github.com/Parabol/action.git/node_modules/graphql/execution/execute.js:688:10)
      at completeValue (/Users/jordanhusney/Source/Repositories/github.com/Parabol/action.git/node_modules/graphql/execution/execute.js:597:12)
      at completeValueWithLocatedError (/Users/jordanhusney/Source/Repositories/github.com/Parabol/action.git/node_modules/graphql/execution/execute.js:517:21)
      at completeValueCatchingError (/Users/jordanhusney/Source/Repositories/github.com/Parabol/action.git/node_modules/graphql/execution/execute.js:493:21)
      at /Users/jordanhusney/Source/Repositories/github.com/Parabol/action.git/node_modules/graphql/execution/execute.js:620:25
      at Array.forEach (native)
      at forEach (/Users/jordanhusney/Source/Repositories/github.com/Parabol/action.git/node_modules/iterall/index.js:272:25)
      at completeListValue (/Users/jordanhusney/Source/Repositories/github.com/Parabol/action.git/node_modules/graphql/execution/execute.js:616:24)
      at completeValue (/Users/jordanhusney/Source/Repositories/github.com/Parabol/action.git/node_modules/graphql/execution/execute.js:580:12)
      at completeValueWithLocatedError (/Users/jordanhusney/Source/Repositories/github.com/Parabol/action.git/node_modules/graphql/execution/execute.js:517:21)
      at completeValueCatchingError (/Users/jordanhusney/Source/Repositories/github.com/Parabol/action.git/node_modules/graphql/execution/execute.js:493:21)
      at resolveField (/Users/jordanhusney/Source/Repositories/github.com/Parabol/action.git/node_modules/graphql/execution/execute.js:461:10)
      at /Users/jordanhusney/Source/Repositories/github.com/Parabol/action.git/node_modules/graphql/execution/execute.js:273:18
      at Array.reduce (native)
      at executeFields (/Users/jordanhusney/Source/Repositories/github.com/Parabol/action.git/node_modules/graphql/execution/execute.js:270:42)
      at completeObjectValue (/Users/jordanhusney/Source/Repositories/github.com/Parabol/action.git/node_modules/graphql/execution/execute.js:688:10)
      at completeValue (/Users/jordanhusney/Source/Repositories/github.com/Parabol/action.git/node_modules/graphql/execution/execute.js:597:12)
      at completeValueWithLocatedError (/Users/jordanhusney/Source/Repositories/github.com/Parabol/action.git/node_modules/graphql/execution/execute.js:517:21)
      at completeValueCatchingError (/Users/jordanhusney/Source/Repositories/github.com/Parabol/action.git/node_modules/graphql/execution/execute.js:493:21)
      at /Users/jordanhusney/Source/Repositories/github.com/Parabol/action.git/node_modules/graphql/execution/execute.js:620:25
      at Array.forEach (native)
      at forEach (/Users/jordanhusney/Source/Repositories/github.com/Parabol/action.git/node_modules/iterall/index.js:272:25)
      at completeListValue (/Users/jordanhusney/Source/Repositories/github.com/Parabol/action.git/node_modules/graphql/execution/execute.js:616:24)
      at completeValue (/Users/jordanhusney/Source/Repositories/github.com/Parabol/action.git/node_modules/graphql/execution/execute.js:580:12)
      at completeValueWithLocatedError (/Users/jordanhusney/Source/Repositories/github.com/Parabol/action.git/node_modules/graphql/execution/execute.js:517:21)
      at completeValueCatchingError (/Users/jordanhusney/Source/Repositories/github.com/Parabol/action.git/node_modules/graphql/execution/execute.js:493:21)
      at resolveField (/Users/jordanhusney/Source/Repositories/github.com/Parabol/action.git/node_modules/graphql/execution/execute.js:461:10)
      at /Users/jordanhusney/Source/Repositories/github.com/Parabol/action.git/node_modules/graphql/execution/execute.js:273:18
      at Array.reduce (native)
      at executeFields (/Users/jordanhusney/Source/Repositories/github.com/Parabol/action.git/node_modules/graphql/execution/execute.js:270:42)
      at completeObjectValue (/Users/jordanhusney/Source/Repositories/github.com/Parabol/action.git/node_modules/graphql/execution/execute.js:688:10)
      at completeValue (/Users/jordanhusney/Source/Repositories/github.com/Parabol/action.git/node_modules/graphql/execution/execute.js:597:12)
      at /Users/jordanhusney/Source/Repositories/github.com/Parabol/action.git/node_modules/graphql/execution/execute.js:554:14
    message: 'Expected a value of type "ProjectStatus" but received: active',
    locations: [ [Object] ],
    path: [ 'summarizeMeeting', 'invitees', 0, 'projects', 0, 'status' ] },
  { Error: Expected a value of type "ProjectStatus" but received: active
      at completeLeafValue (/Users/jordanhusney/Source/Repositories/github.com/Parabol/action.git/node_modules/graphql/execution/execute.js:639:11)
      at completeValue (/Users/jordanhusney/Source/Repositories/github.com/Parabol/action.git/node_modules/graphql/execution/execute.js:586:12)
      at completeValueWithLocatedError (/Users/jordanhusney/Source/Repositories/github.com/Parabol/action.git/node_modules/graphql/execution/execute.js:517:21)
      at completeValueCatchingError (/Users/jordanhusney/Source/Repositories/github.com/Parabol/action.git/node_modules/graphql/execution/execute.js:493:21)
      at resolveField (/Users/jordanhusney/Source/Repositories/github.com/Parabol/action.git/node_modules/graphql/execution/execute.js:461:10)
      at /Users/jordanhusney/Source/Repositories/github.com/Parabol/action.git/node_modules/graphql/execution/execute.js:273:18
      at Array.reduce (native)
      at executeFields (/Users/jordanhusney/Source/Repositories/github.com/Parabol/action.git/node_modules/graphql/execution/execute.js:270:42)
      at completeObjectValue (/Users/jordanhusney/Source/Repositories/github.com/Parabol/action.git/node_modules/graphql/execution/execute.js:688:10)
      at completeValue (/Users/jordanhusney/Source/Repositories/github.com/Parabol/action.git/node_modules/graphql/execution/execute.js:597:12)
      at completeValueWithLocatedError (/Users/jordanhusney/Source/Repositories/github.com/Parabol/action.git/node_modules/graphql/execution/execute.js:517:21)
      at completeValueCatchingError (/Users/jordanhusney/Source/Repositories/github.com/Parabol/action.git/node_modules/graphql/execution/execute.js:493:21)
      at /Users/jordanhusney/Source/Repositories/github.com/Parabol/action.git/node_modules/graphql/execution/execute.js:620:25
      at Array.forEach (native)
      at forEach (/Users/jordanhusney/Source/Repositories/github.com/Parabol/action.git/node_modules/iterall/index.js:272:25)
      at completeListValue (/Users/jordanhusney/Source/Repositories/github.com/Parabol/action.git/node_modules/graphql/execution/execute.js:616:24)
      at completeValue (/Users/jordanhusney/Source/Repositories/github.com/Parabol/action.git/node_modules/graphql/execution/execute.js:580:12)
      at completeValueWithLocatedError (/Users/jordanhusney/Source/Repositories/github.com/Parabol/action.git/node_modules/graphql/execution/execute.js:517:21)
      at completeValueCatchingError (/Users/jordanhusney/Source/Repositories/github.com/Parabol/action.git/node_modules/graphql/execution/execute.js:493:21)
      at resolveField (/Users/jordanhusney/Source/Repositories/github.com/Parabol/action.git/node_modules/graphql/execution/execute.js:461:10)
      at /Users/jordanhusney/Source/Repositories/github.com/Parabol/action.git/node_modules/graphql/execution/execute.js:273:18
      at Array.reduce (native)
      at executeFields (/Users/jordanhusney/Source/Repositories/github.com/Parabol/action.git/node_modules/graphql/execution/execute.js:270:42)
      at completeObjectValue (/Users/jordanhusney/Source/Repositories/github.com/Parabol/action.git/node_modules/graphql/execution/execute.js:688:10)
      at completeValue (/Users/jordanhusney/Source/Repositories/github.com/Parabol/action.git/node_modules/graphql/execution/execute.js:597:12)
      at completeValueWithLocatedError (/Users/jordanhusney/Source/Repositories/github.com/Parabol/action.git/node_modules/graphql/execution/execute.js:517:21)
      at completeValueCatchingError (/Users/jordanhusney/Source/Repositories/github.com/Parabol/action.git/node_modules/graphql/execution/execute.js:493:21)
      at /Users/jordanhusney/Source/Repositories/github.com/Parabol/action.git/node_modules/graphql/execution/execute.js:620:25
      at Array.forEach (native)
      at forEach (/Users/jordanhusney/Source/Repositories/github.com/Parabol/action.git/node_modules/iterall/index.js:272:25)
      at completeListValue (/Users/jordanhusney/Source/Repositories/github.com/Parabol/action.git/node_modules/graphql/execution/execute.js:616:24)
      at completeValue (/Users/jordanhusney/Source/Repositories/github.com/Parabol/action.git/node_modules/graphql/execution/execute.js:580:12)
      at completeValueWithLocatedError (/Users/jordanhusney/Source/Repositories/github.com/Parabol/action.git/node_modules/graphql/execution/execute.js:517:21)
      at completeValueCatchingError (/Users/jordanhusney/Source/Repositories/github.com/Parabol/action.git/node_modules/graphql/execution/execute.js:493:21)
      at resolveField (/Users/jordanhusney/Source/Repositories/github.com/Parabol/action.git/node_modules/graphql/execution/execute.js:461:10)
      at /Users/jordanhusney/Source/Repositories/github.com/Parabol/action.git/node_modules/graphql/execution/execute.js:273:18
      at Array.reduce (native)
      at executeFields (/Users/jordanhusney/Source/Repositories/github.com/Parabol/action.git/node_modules/graphql/execution/execute.js:270:42)
      at completeObjectValue (/Users/jordanhusney/Source/Repositories/github.com/Parabol/action.git/node_modules/graphql/execution/execute.js:688:10)
      at completeValue (/Users/jordanhusney/Source/Repositories/github.com/Parabol/action.git/node_modules/graphql/execution/execute.js:597:12)
      at /Users/jordanhusney/Source/Repositories/github.com/Parabol/action.git/node_modules/graphql/execution/execute.js:554:14
    message: 'Expected a value of type "ProjectStatus" but received: active',
    locations: [ [Object] ],
    path: [ 'summarizeMeeting', 'invitees', 3, 'projects', 0, 'status' ] } ]
Warning: Unknown prop `align` on <table> tag. Remove this prop from the element. For details, see https://fb.me/react-unknown-prop
    in table
    in Layout
Warning: Failed prop type: Always include the `height` and `width` attributes of your <img> as integers. See https://www.campaignmonitor.com/resources/will-it-work/guidelines/
    in Img
Warning: Failed prop type: Relative src URLs can break (i.e. if recipients are outside the company network) and make your content unavailable to view.
    in Img
Warning: Unknown props `border`, `tagName` on <table> tag. Remove these props from the element. For details, see https://fb.me/react-unknown-prop
    in table (created by Table)
    in Table
mailgun: no API key, so not sending the following:
    From: 
    To: , taya@prbl.co, terry@prbl.co, matt@prbl.co
    Subject: Action Meeting Summary from Parabol
    Body: Hello Parabol! As a team you discussed 1 Agenda Items 
resulting in 2 New Projects 
and 0 New Actions.
```

- **OS:** OSX
- **Browser:** Chrome

### LINKED ISSUE #779: Exception on new (test) team member invitation [dnd-sortOrder]
## Issue - Bug

This might be something specific to how we're initializing our mock users for development. I'm using `jordan@prbl.io`.

Repro steps:

1. Login as `jordan@prbl.co`
1. Click on `Engineering` team
1. Click on `Team Settings`
1. Invite a new users, such as `scott@prbl.co`

Observe exception:

```
DEBUG GraphQL Error: [ { ReqlRuntimeError: Cannot perform bracket on a non-object non-sequence `null` in:
  r.expr(["scott@prbl.co"]).do(function(var_29) {
      return {
          inviteEmails: r.table("Invitation").getAll(r.args(var_29), {
              index: "email"
          }).filter(function(var_30) {
              return var_30("tokenExpiration").ge(r.epochTime(1488745645308))
          })("email").coerceTo("array"),
          teamMembers: r.table("TeamMember").getAll("team456", {
              index: "teamId"
          }).merge(function(var_31) {
              return {
                  userOrgs: r.table("User").get(var_31("userId"))("userOrgs")
                            ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^            
              }
          }).coerceTo("array")
      }
  })
  
      at Connection._processResponse (/Users/jordanhusney/Source/Repositories/github.com/Parabol/action.git/node_modules/rethinkdbdash/lib/connection.js:395:15)
      at Socket.<anonymous> (/Users/jordanhusney/Source/Repositories/github.com/Parabol/action.git/node_modules/rethinkdbdash/lib/connection.js:201:14)
      at emitOne (events.js:96:13)
      at Socket.emit (events.js:189:7)
      at Socket.EventEmitter.emit (/Users/jordanhusney/Source/Repositories/github.com/Parabol/action.git/node_modules/sc-domain/index.js:12:31)
      at readableAddChunk (_stream_readable.js:176:18)
      at Socket.Readable.push (_stream_readable.js:134:10)
      at TCP.onread (net.js:551:20)
  From previous event:
      at Function.Term.run (/Users/jordanhusney/Source/Repositories/github.com/Parabol/action.git/node_modules/rethinkdbdash/lib/term.js:142:15)
      at Function.Term.then (/Users/jordanhusney/Source/Repositories/github.com/Parabol/action.git/node_modules/rethinkdbdash/lib/term.js:2876:15)
    message: 'Cannot perform bracket on a non-object non-sequence `null` in:\nr.expr(["scott@prbl.co"]).do(function(var_29) {\n    return {\n        inviteEmails: r.table("Invitation").getAll(r.args(var_29), {\n            index: "email"\n        }).filter(function(var_30) {\n            return var_30("tokenExpiration").ge(r.epochTime(1488745645308))\n        })("email").coerceTo("array"),\n        teamMembers: r.table("TeamMember").getAll("team456", {\n            index: "teamId"\n        }).merge(function(var_31) {\n            return {\n                userOrgs: r.table("User").get(var_31("userId"))("userOrgs")\n                          ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^            \n            }\n        }).coerceTo("array")\n    }\n})\n',
    locations: [ [Object] ],
    path: [ 'inviteTeamMembers' ] } ]
```

### PR DESCRIPTION
**Pull request by mattkrick**
### Integration tests:
- [x] expect that after a meeting, all new actions & projects to show up at the top of everyone's team & user dashboards
- [x] expect to get a meeting summary when you click end meeting
- [x] expect new items to be created with a little bit of noise so if Team A, B, C all have a single item, in the Done column, then the user can drag C between A and B
- [x] given Project A and B in the done column, drag B to be on top of A, then drag A to be on top of B (this was broken before, yikes!)
- [x] if the resolution of the sort order is so small that the language can't detect a difference, then the drag fails, a card gets displaced (either the one above or below) and trying again succeeds.
- [x] fix #733
other fixer-uppers
- [x] can upload a new avatar image #754 
- [x] can approve a user into an org where the requested team is one that the org leader is not a part of #765 

### Discrete Test Fixes
- [x] fix #778 
- [x] fix #779

### CODE REVIEW

**Comment by jordanh** [comment]
- [ ] expect that after a meeting, all new actions & projects to show up at the top of everyone's team & user dashboards
- [ ] expect to get a meeting summary when you click end meeting
- [ ] expect new items to be created with a little bit of noise so if Team A, B, C all have a single item, in the Done column, then the user can drag C between A and B
- [ ] given Project A and B in the done column, drag B to be on top of A, then drag A to be on top of B (this was broken before, yikes!)
- [ ] if the resolution of the sort order is so small that the language can't detect a difference, then the drag fails, a card gets displaced (either the one above or below) and trying again succeeds.

**Comment by jordanh** [comment]
I gave this a fairly complete once over on the flight to SF tonight. I only was able to find #777; this is looking _sharp_ mate.

**Comment by mattkrick** [comment]
@jordanh don't mind the codecov errors, somehow whatever is on master got there without making a coverage report. the rest should be good to go

File: circle.yml [hunk 1]
```diff
@@ -36,7 +36,7 @@ stages:
         command: |
           apt-get update && apt-get -yq install build-essential
           if [ ! -x "$(which yarn)" ] ; then
-            npm -i -g yarn
```
Comment (jordanh) [inline circle.yml]: Oopies. Good catch!

By the way, I think what is happening here is that some of the CircleCI containers are using old images while some are using new that have yarn. It's real goofy.

File: package.json [hunk 2]
```diff
@@ -30,7 +30,7 @@
     "build:server": "NODE_ENV=production webpack --config ./webpack/server.babel.js",
     "db:migrate": "./src/server/database/migrate.sh up-all",
     "db:migrate-testing": "NODE_ENV=test ./src/server/database/migrate.sh up-all",
-    "test": "jest && codecov",
+    "test": "jest --bail --coverage && codecov",
```
Comment (jordanh) [inline package.json]: Do you like bail as the default?

File: src/server/__tests__/setup/MockDB.js [hunk 3]
```diff
@@ -1,13 +1,46 @@
 import getRethink from 'server/database/rethinkDriver';
 import testUsers from 'server/__tests__/setup/testUsers';
 import shortid from 'shortid';
-import {BILLING_LEADER, LOBBY} from 'universal/utils/constants';
+import {
```
Comment (jordanh) [inline src/server/__tests__/setup/MockDB.js]: This is such a good idea, and worthy of a blog article!

File: src/server/database/migrations/20170228132400-cleanupProjectIndices.js [hunk 4]
```diff
@@ -0,0 +1,48 @@
+exports.up = async(r) => {
+  const fields = [
+    r.table('Project').replace((row) => {
+      return row
+        .merge({
+          sortOrder: row('teamSort').default(r.random())
```
Comment (jordanh) [inline src/server/database/migrations/20170228132400-cleanupProjectIndices.js]: Heh. I like this.

File: package.json (hunk 2 above)
Comment (mattkrick) [inline package.json]: funny enough, it's the only way i can ensure an exit code of 1 on a failure...

Comment (jordanh) [inline package.json]: Ha! Ok. Well, it'll certainly get the job done (failing a CI build...)

**Comment by jordanh** [comment]
The code looks _excellent_. Your testing game is _strong_. I considered going to snapshot route for making comparisons – I'm really glad you took the plunge.

Would you like me to run the integration test plan at the top of this PR?

**Comment by mattkrick** [comment]
@jordanh yeah, those top 8 checkboxes are all for you! maybe i'm crazy, but the DnD _feels_ faster, i could just be crazy though

**Comment by jordanh** [comment]
No, it _does_ feel faster. I'm not sure why. I'm on it now...

**Comment by jordanh** [comment]
Getting super, duper close here. I ran into a couple of exceptions which may or may not be related to the test environment. Mind having a look at them? I'll be here to help and retest.

Also, @ackernaut, I am going to change the copy from "Upload Avatar" to "Choose Photo." When I see "Upload Avatar" I keep thinking, "but I haven't told you what I want to upload yet!"

**Comment by jordanh** [comment]
Running another test pass now!

**Comment by jordanh** [comment]
## Integration Test Results

All tests passed.

We're GTG now. Do you want the honor of merging it sir?

**Pull request merged**